lib/scheduler.js

```diff
@@ -172,19 +175,22 @@ class Scheduler {
         }
 
         if (schedule.time) {
-            let start = schedule.time.start;
-            let times;
-            if (start === 'sunrise' || start === 'sunset') {
-                times = this.suncalc.getTimes(new Date(), this.latitude, this.longitude);
-                start = times[start];
+            let start = schedule.time.start,
+                times,
+                astroEvent,
+                now = new this.Date();
+            now.setHours(12, 0, 0, 0);
+            if (start && (astroEvent = this._isAstro(start))) {
```

Comment:
Better move it up to the variable declaration and please do not reuse the variable name later with a diufferent meaning ... There is no problem in having two variables :-)

Reply:
also below in the further changed lines

Reply:
not sure if i understood well. I also renamed `now` to the more meaningful `todayNoon`

Reply:
I mean to not use an assignment in an if.

use

```
             astroEvent = this._isAstro(start);
             if (start && astroEvent) { 
```